daemon/xvcpico.c

```diff
@@ -194,12 +195,12 @@ static int sread(int fd, void *target, int len) {
   return 1;
 }
 
-int handle_data(int fd) {
+int handle_data(int fd, int buffer_size, int verbose) {
   uint32_t len, nr_bytes;
 
   do {
     char cmd[16];
-    unsigned char buffer[BUFFER_SIZE], result[BUFFER_SIZE / 2];
+    unsigned char buffer[buffer_size], result[buffer_size / 2];
```

Comment:
I just realized that we may not be able to create large buffers on the stack safely.

I recommend moving to dynamically allocated buffers using `malloc`. I can also get this done in next few weeks (testing is the hard part).

Reply:
Ok You are right. It is better to use dynamically allocate buffers with malloc